### ISSUE TITLE
Add QuSpin backend for ED calculations

### ISSUE THREAD

**Issue by PhilipVinc** [netket/netket#1941]
QuSpin will soon become easy to install 
https://github.com/QuSpin/QuSpin/pull/703

We should add some simple methods to convert our operators to their format.

A major thing that should be prototyped is a way to target a particular symmetry subsector, because we do not currently store this information in the Hilbert space.
Could be added to the conversion method?
Also raises an interesting question about whether the Hilbert space could have this constraint. 
(though right now our Hilbert space is more of a computational basis than anything else..)

**Comment by gcarleo** [comment]
we already have a primitive way of enforcing the symmetry sector in the Hilbert space, right? (when you fix the total magnetization to a specific value, say) the question is to extend it to other constraints ?

**Comment by PhilipVinc** [comment]
yeah. I was more thinking of lattice symmetries (Translation & point group).

**Comment by PhilipVinc** [comment]
Which is also something worth thinking about, because while we support those things the way to identify which character you want to pass to the simmetrization layer depending on what eigenvalue you target is not trivial...
